lib/hooks/userspace/hook-redis.js

```diff
@@ -32,6 +32,23 @@ function createClientWrap(createClient) {
   };
 }
 
+// Used for redis version > 2.3
+function createStreamWrap(create_stream) {
+  return function create_stream_trace() {
+    if (!this.stream) {
+      Object.defineProperty(this, 'stream', {
```

Comment:
The issue with wrapping the constructor is that before constructor invocation, the `this` object isn't accessible to my knowledge and after constructor invocation, it is already too late and create_streams has been called (https://github.com/NodeRedis/node_redis/blob/master/index.js#L109). This was the only way that I could think of to get a handle on `this` before the constructor finished executing.